### ISSUE TITLE
Expand make variables in vars

### ISSUE THREAD

**Issue by mariusgrigoriu** [bazelbuild/rules_jsonnet#39]
I'm trying to inject make variables into the template expansion to pass variables in from the bazel command line. This is useful when there are external environmental values, such as host names, in the context of Kubernetes templating.

For example:
`bazel build :k8s --define fqdn=example.com`

```
jsonnet_to_json(
    name = "k8s",
    src = "k8s.jsonnet",
    outs = ["k8s.json"],
    vars = {"fqdn": "$(fqdn)"},
)
```

**Comment by Globegitter** [comment]
@mariusgrigoriu Just started out with using jsonnet for kubernetes manifests and we would have similar requirements in regard to ingress host names. I am a fan of having everything checked into the repository so my idea was making use of `--code-file/file` (not implemented here yet but see https://github.com/google/jsonnet/pull/110 for usage) and then define it as something like:

```
jsonnet_to_json(
    name = "k8s",
    src = "k8s.jsonnet",
    outs = ["k8s.json"],
    file = select({
    "//common/config:namespace_dev": {"fqdn": ":fqdn_dev.txt"},
    "//common/config:namespace_something": {"fqdn": ":fqdn_something.txt"},
    }),
)
```
So one would run e.g. `bazel build :k8s --define namespace=dev` and that would then make it select the right checked in file which contains the fqdn.

Your approach is of course total valid as well and my thought of approach has it's own drawbacks but since I just thought of that yesterday/today and wanted to look into it in the near future I thought I'd mention it here.

**Comment by mariusgrigoriu** [comment]
I think that approach is better if you have a limited number of environments. In our case, the name is dynamically generated from earlier in the build process. Each pipeline run spins up a whole cluster with a unique name. Normally I'd use stamp variables for something like that, but I'm integrating with a complicated make system while incrementally converting to Bazel. (I'm also new to Bazel so I'm sure there are better ways of accomplishing this that I don't know about.)

**Comment by sparkprime** [comment]
Is this backwards incompatible for people who were previously using $ in that string and expecting it to be passed through to Jsonnet?

**Comment by mariusgrigoriu** [comment]
A $ alone should be fine, however $( ) would be interpreted as a make variable. People who have those would need to use $$( )

**Comment by mariusgrigoriu** [comment]
Merged in #40 
Thank you.
